v1.0 Release Candidate

chris-cheshire opened the pull request.

maxulysse commented:
I think all test related data should be in the `nf-core/test-datasets` repo

maxulysse reviewed (dismissed):
Really like what you did, I think there's a couple of things to update, but we should be pretty close to a release

maxulysse commented on .nf-core.yaml:
```diff
@@ -0,0 +1,10 @@
+  - docs/README.md
+  - bin/markdown_to_html.py
+files_exist:
+  - .github/markdownlint.yml
```
Did you try removing that?

maxulysse commented on assets/multiqc_config.yaml:
```diff
@@ -9,3 +9,109 @@ report_section_order:
+        - It can be due to the Tn5 preference.
+        - What you might be detecting is the 10-bp periodicity that shows up as a sawtooth pattern in the length distribution. If so, this is normal and will not affect alignment or peak calling. In any case we do not recommend trimming as the bowtie2 parameters that we list will give accurate mapping information without trimming.
+
+# fastqc_config:
+#     fastqc_theoretical_gc: "hg38_genome"
+
+# extra_fn_clean_exts:
+#   - ".spikein.bowtie2"
+#   - "_processedFile"
+# extra_fn_clean_trim:
+#   - "#"
+#   - ".myext"
+
+# Customise the module search patterns to speed up execution time
+#  - Skip module sub-tools that we are not interested in
+#  - Replace file-content searching with filename pattern searching
+#  - Don't add anything that is the same as the MultiQC default
+# See https://multiqc.info/docs/#optimise-file-search-patterns for details
```
Can remove comments

chris-cheshire replied:
Done

maxulysse commented on conf/laptop.config:
```diff
@@ -0,0 +1,11 @@
+/*
```
I'm still not sure we should have such a configuration here actually

mashehu replied:
I agree, you should untrack this file.

maxulysse commented on modules/local/get_chrom_sizes.nf:
```diff
@@ -0,0 +1,39 @@
+    script:
+    def software = 'samtools'
+    """
+    samtools faidx $fasta
+    cut -f 1,2 ${fasta}.fai > ${fasta}.sizes
+    echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//' > ${software}.version.txt
```
I think you can make it as a nf-core/modules, it's an easy command, but could be worth it to share with others

chris-cheshire replied:
Ok I will get this submitted

maxulysse commented on .gitignore:
```diff
@@ -3,7 +3,22 @@ work/
+dev/docker/static_reports/test_data/*
+dev/docker/static_reports/test_output/*
+!dev/docker/static_reports/test_data/_dummy.txt
+!dev/docker/static_reports/test_output/_dummy.txt
```
Why escaping this file if not in the repo?

chris-cheshire replied:
Yep, I have cleaned up this file now

maxulysse reviewed (commented):
There are a couple of comments left by the reviewers that are not yet addressed.
Apart from that, it's looking super good.

chris-cheshire commented:
@maxulysse I cant see the unaddressed comments? thanks again for your review - I have resolved your comments. I will submit the get_chrom_sizes module, but if poss I would like to press ahead with release rather than waiting for that to be accepted into the modules repo.

drpatelh reviewed (approved):
Great work @chris-cheshire ! 😍

mashehu commented on docs/output.md:
```diff
@@ -1,35 +1,71 @@
-* [FastQC](#fastqc) - Raw read QC
-* [MultiQC](#multiqc) - Aggregate report describing results and QC from the whole pipeline
-* [Pipeline information](#pipeline-information) - Report metrics generated during the workflow execution
+- [nf-core/cutandrun: Output](#nf-corecutandrun-output)
+    - [Introduction](#introduction)
+    - [Pipeline overview](#pipeline-overview)
+    - [Preprocessing](#preprocessing)
+        - [Samplesheet check](#samplesheet-check)
+        - [Fastq merging](#fastq-merging)
+        - [FastQC](#fastqc)
+        - [TrimGalore](#trimgalore)
+    - [Alignment](#alignment)
+        - [Bowtie 2](#bowtie-2)
+    - [Alignment post-processing](#alignment-post-processing)
+        - [samtools](#samtools)
+        - [picard MarkDuplicates/RemoveDuplicates](#picard-markduplicatesremoveduplicates)
+    - [Peak Calling](#peak-calling)
+        - [Bam to bedgraph](#bam-to-bedgraph)
+        - [Clip bedfiles](#clip-bedfiles)
+        - [Bed to bigwig](#bed-to-bigwig)
+        - [SEACR peak calling](#seacr-peak-calling)
+        - [BEDtools](#bedtools)
+    - [Reporting](#reporting)
+        - [Python reporting](#python-reporting)
+        - [MultiQC](#multiqc)
+        - [IGV](#igv)
+        - [Deeptools](#deeptools)
+    - [Workflow reporting and genomes](#workflow-reporting-and-genomes)
+        - [Reference genome files](#reference-genome-files)
+        - [Pipeline information](#pipeline-information)
```
No need for a markdown based ToC anymore. It will be generated from the headings on the website automatically
